### PR TITLE
Features/publish geoarea mutation

### PR DESCRIPTION
**Pull request by sudan45**
## Addresses 
- https://github.com/the-deep/server/issues/1453

## Depends
- https://github.com/the-deep/server/issues/1449

Mention related users here if any.

## This PR doesn't introduce any:

- [ ] temporary files, auto-generated files or secret keys
- [ ] n+1 queries
- [ ] flake8 issues
- [ ] `print`
- [ ] typos
- [ ] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

**Comment by thenav56** [comment]
@sudan45 This includes a lot of changes. Can you check why?

File: apps/geo/serializers.py [hunk 1]
```diff
@@ -148,3 +148,65 @@ class Meta:
             'region_title', 'admin_level_level', 'admin_level_title',
             'parent'
         )
+
+
+class RegionGqSerializer(ProjectPropertySerializerMixin, UserResourceSerializer):
+    project = serializers.IntegerField()
```
Comment (susilnem) [inline apps/geo/serializers.py]: Let's use [PrimaryKeyRelatedField](https://www.django-rest-framework.org/api-guide/relations/#primarykeyrelatedfield) here.

File: apps/geo/serializers.py [hunk 2]
```diff
@@ -148,3 +148,65 @@ class Meta:
...
+            project.regions.add(region)
+
+        return region
+
+    def update(self, validated_data, instance):
+        return True
+
+
+class PublishRegionGqSerializer(ProjectPropertySerializerMixin, UserResourceSerializer):
+    project = serializers.IntegerField()
```
Comment (susilnem) [inline apps/geo/serializers.py]: Here too.

File: apps/geo/tests/test_mutations.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        minput = dict(
+            project=project.id,
+            code="NPL",
+            title="Test",
+        )
+        # without login
+        _query_check(minput, assert_for_error=True)
+
+        self.force_login(user)
```
Comment (susilnem) [inline apps/geo/tests/test_mutations.py]: Add test for different permissions too.

**Comment by thenav56** [comment]
@sudan45 Let's rebase this

**Review by susilnem — COMMENTED** [review]
LGTM!

**Comment by sudan45** [comment]
This branch has been merged with `features/add-region-mutation`

**Comment by sudan45** [comment]
Code merge this branch
- https://github.com/the-deep/server/pull/1449

**Pull request closed without merging**